browser/resources/settings/brave_leo_assistant_page/model_config_ui.ts

```diff
@@ -0,0 +1,173 @@
+import * as mojom from './brave_leo_assistant_browser_proxy.js'
+import { BaseMixin } from '../base_mixin.js'
+import { getTemplate } from './model_config_ui.html.js'
+
+const ModelConfigUIBase = PrefsMixin(I18nMixin(BaseMixin(PolymerElement)))
+
+export class ModelConfigUI extends ModelConfigUIBase {
+  static get is() {
+    return 'model-config-ui'
+  }
+
+  static get template() {
+    return getTemplate()
+  }
+
+  static get properties() {
+    return {
+      label: {
+        type: String
+      },
+      modelRequestName: {
+        type: String
+      },
+      endpointUrl: {
+        type: String
+      },
+      apiKey: {
+        type: String,
+        value: ''
+      },
+      isUrlInvalid: {
+        type: Boolean,
+        value: false,
+      },
+      invalidUrlErrorMessage: {
+        type: String,
+        value: ''
+      },
+      modelItem: {
+        type: Object,
+        value: null,
+        notify: true,
+        observer: 'onModelItemChange_'
+      },
+      isEditing_: {
+        type: Boolean,
+        value: false,
+        readOnly: true,
+        computed: 'computeIsEditing_(modelItem)'
+      },
+      buttonLabel_: {
+        type: String,
+        computed: 'computeButtonLabel_(isEditing_)'
+      }
+    }
+  }
+
+  label: string
+  modelRequestName: string
+  endpointUrl: string
+  apiKey: string
+  modelItem: mojom.Model | null
+  isEditing_: boolean
+  isUrlInvalid: boolean
+  invalidUrlErrorMessage: string
+
+  override ready() {
+    super.ready()
+  }
+
+  handleClick_() {
+    if (!this.saveEnabled_()) {
+      return
+    }
+
+    const mojomUrl = { url: '' }
+    mojomUrl.url = this.endpointUrl
+
+    // Send empty string if adding new model
+    const modelKey = this.isEditing_ ? this.modelItem?.key : ''
+
+    if (modelKey === undefined) {
+      console.error('Model key is undefined')
+      return
+    }
+
+    const modelConfig: mojom.Model = {
+      options: {
+        customModelOptions: {
+          modelRequestName: this.modelRequestName,
+          endpoint: mojomUrl,
+          apiKey: this.apiKey
+        }
+      },
+      key: modelKey,
+      displayName: this.label
+    }
+
+    this.fire('save', { modelConfig })
+  }
+
+  handleCloseClick_() {
+    this.fire('close')
+  }
+
+  onModelLabelChange_(e: any) {
+    this.label = e.target.value
+  }
+
+  onModelRequestNameChange_(e: any) {
+    this.modelRequestName = e.target.value
+  }
+
+  onModelServerEndpointChange_(e: any) {
+    this.endpointUrl = e.target.value
+
+    // We need to check if the URL is valid because sending bad URL will cause
+    // renderer to crash. This is mainly due to mojo IPC not being able to
+    // handle bad URLs properly for |mojomUrl| type
+    try {
+      new URL(e.target.value)
+      this.isUrlInvalid = false
+    } catch {
+      this.isUrlInvalid = true
+      this.invalidUrlErrorMessage = this.i18n(
+        'braveLeoAssistantEndpointInvalidError'
+      )
```

Comment:
nit: still don't understand why this isn't just set on the html attribute directly, but not a big concern at all